### PR TITLE
My portal automation

### PR DESCRIPTION
**Pull request by sunnysoomro12**

### CODE REVIEW

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 1]
```diff
@@ -0,0 +1,868 @@
...
+        Assert.assertEquals(driver.findElement(By.id("step-2")).getCssValue("background-color"),"rgba(0, 150, 219, 1)");
+        Assert.assertEquals(driver.findElement(By.id("step-1")).getCssValue("background-color"),"rgba(0, 92, 132, 1)");
+        Assert.assertEquals(driver.findElement(By.id("step-3")).getCssValue("background-color"),"rgba(190, 190, 190, 1)");
+    }
+
+    @Then("^Can not accept non numeric values$")
+    public void can_not_accept_non_numeric_values() throws InterruptedException {
+        objSmartSegPageElements = new SmartSegPageElements((RemoteWebDriver) driver);
+        Thread.sleep(1000);
+        Actions onetime = new Actions (driver);
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: could you please add the actions class to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 2]
```diff
@@ -0,0 +1,868 @@
...
+        driver.findElement(By.xpath("//*[@id='root']/div/div[3]/div/main/div[2]/div/div/div[2]/div/form/div[2]/div/section/div[2]/div[2]/div/h4")).click();
+        //Validate the field is not accepting non-numeric digits
+        String totalAmount = driver.findElement(By.xpath("//*[@data-id='txtTotalAmount']")).getText();
+        Assert.assertEquals(totalAmount,"$NaN");
+    }
+    @Then("^Validate deposit detail page features$")
+    public void validate_deposit_detail_page_features() throws InterruptedException {
+        objSmartSegPageElements = new SmartSegPageElements((RemoteWebDriver) driver);
+        Thread.sleep(1000);
+        Actions onetime = new Actions (driver);
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: could you please add the actions class to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 3]
```diff
@@ -0,0 +1,868 @@
...
+        // Validate Add new fund PWS-2009
+        fundOption.selectByVisibleText("Front-end");
+        Select fundname = new Select(driver.findElement(By.name("newFundDirections.0.fundCode")));
+        fundname.selectByVisibleText("BOND GIF");
+        Select FEOption = new Select(driver.findElement(By.name("newFundDirections.0.frontEndPercentage")));
+        FEOption.selectByVisibleText("3%");
+        driver.findElement(By.name("newFundDirections.0.fundDepositAmount")).sendKeys("50");
+        // Add fund 2
+        objSmartSegPageElements.addNewFundButtonClick();
+        Select fundOption2 = new Select(driver.findElement(By.name("newFundDirections.1.feeOption")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: could you please add the actions class to the page factory

**Comment by ShiviPrabhakar** [comment]
@sunnysoomro12 anything all the actions classes and Select all of that has to be in page factory

File: src/main/java/empire/StepDef/Portal/Retail/AdvisorPortal/SmartTransfer/smartTransfer.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+    }
+
+    @Given("^Add (.*?) and (.*?) new$")
+    public void add_username_and_password_new(String username, String password) throws InterruptedException {
+        objSmartTrasferPageElements = new SmartTransferPageElements((RemoteWebDriver) driver);
+        objSmartTrasferPageElements.username(username);
+        objSmartTrasferPageElements.password(password);
+        objSmartTrasferPageElements.SubmitBtnCLick();
+        Thread.sleep(15000);
```
Comment (ShiviPrabhakar) [inline src/main/java/empire/StepDef/Portal/Retail/AdvisorPortal/SmartTransfer/smartTransfer.java]: as @kevin-benoit suggested try the wait time as he suggested

File: src/main/java/empire/StepDef/Portal/Retail/CustomerPortal/customerportal.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        Assert.assertEquals(driver.findElement(By.className("greeting-container")).getText(),"Hello");
+
+
+
+    }
+
+        @Then("^First quick link is Go Paperless$")
+        public void first_quick_link_is_Go_Paperless() throws InterruptedException {
+
+            Thread.sleep(15000);
```
Comment (ShiviPrabhakar) [inline src/main/java/empire/StepDef/Portal/Retail/CustomerPortal/customerportal.java]: try the waiting for the next element that you are waiting for

File: src/main/java/empire/StepDef/Portal/Retail/CustomerPortal/customerportal.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            //System.out.println(driver.findElement(By.className("quick-links-container")).getText());
+            // Validate Go Paperless is on the top of Quick Links
+/*
+            quicklinkContainer = new ArrayList();
+            goPaperless = new ArrayList();
+
+            quicklinkContainer.add("QUICK LINKS");
+            quicklinkContainer.add("Go Paperless");
```
Comment (ShiviPrabhakar) [inline src/main/java/empire/StepDef/Portal/Retail/CustomerPortal/customerportal.java]: remove the commented code

File: src/main/java/empire/StepDef/Portal/Retail/AdvisorPortal/SmartTransfer/smartTransfer.java (hunk 4 above)
Comment (sunnysoomro12) [reply src/main/java/empire/StepDef/Portal/Retail/AdvisorPortal/SmartTransfer/smartTransfer.java]: As per suggestion, code is not working. So I backed up. I have to see, how is your working.

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 3 above)
Comment (sunnysoomro12) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: As we decided, in this stage we dont need to maintain page factory. My 50% code is not in the page factopry

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 2 above)
Comment (sunnysoomro12) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: As we decided, in this stage we dont need to maintain page factory. My 50% code is not in the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 1 above)
Comment (sunnysoomro12) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: As we decided, in this stage we dont need to maintain page factory. My 50% code is not in the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 7]
```diff
@@ -0,0 +1,868 @@
...
+    public void portal_Login_Page() throws InterruptedException {
+        driver.get(Constant.PortalURL);
+    }
+    @Given("^Add (.*?) and (.*?) new$")
+    public void add_username_and_password_new(String username , String password) throws InterruptedException {
+        objSmartSegPageElements = new SmartSegPageElements((RemoteWebDriver) driver);
+        objSmartSegPageElements.username(username);
+        objSmartSegPageElements.password(password);
+        objSmartSegPageElements.SubmitBtnCLick();
+        Thread.sleep(15000);
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: please wait for the next element to remove the thread

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 1 above)
Comment (ShiviPrabhakar) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: move the actions class and select to page factory please rest is fine

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 7 above)
Comment (sunnysoomro12) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: I tried and did not worked

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 1 above)
Comment (sunnysoomro12) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: ok

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java (hunk 2 above)
Comment (ShiviPrabhakar) [reply src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Actions classes and Select classes HAVE to be in page factory looks really unprofessional code wise

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 8]
```diff
@@ -0,0 +1,868 @@
...
+        driver.findElement(By.xpath("//*[@data-id='txtTotalAmount']")).click();
+        driver.findElement(By.name("depositAmount")).click();
+        driver.findElement(By.name("depositAmount")).sendKeys("10000");
+        driver.findElement(By.xpath("//*[@id='root']/div/div[3]/div/main/div[2]/div/div/div[2]/div/form/div[2]/div/section/div[2]/div[2]/div/h4")).click();
+        objSmartSegPageElements.addNewFundButtonClick();
+        Select fundOption = new Select(driver.findElement(By.name("newFundDirections.0.feeOption")));
+        fundOption.selectByVisibleText("Front-end");
+        WebElement FE = driver.findElement(By.name("newFundDirections.0.frontEndPercentage"));
+        fundOption.selectByVisibleText("Front-end");
+        Select fundname = new Select(driver.findElement(By.name("newFundDirections.0.fundCode")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 9]
```diff
@@ -0,0 +1,868 @@
...
+        fundname.selectByVisibleText("BOND GIF");
+        Select FEOption = new Select(driver.findElement(By.name("newFundDirections.0.frontEndPercentage")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 10]
```diff
@@ -0,0 +1,868 @@
...
+        FEOption.selectByVisibleText("3%");
+        driver.findElement(By.name("newFundDirections.0.fundDepositAmount")).sendKeys("50");
+        objSmartSegPageElements.addNewFundButtonClick();
+        Select fundOption2 = new Select(driver.findElement(By.name("newFundDirections.1.feeOption")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 11]
```diff
@@ -0,0 +1,868 @@
...
+        fundOption2.selectByVisibleText("Front-end");
+        Select fundname2 = new Select(driver.findElement(By.name("newFundDirections.1.fundCode")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 12]
```diff
@@ -0,0 +1,868 @@
...
+        fundname2.selectByVisibleText("INCOME GIF");
+        Select FEOption2 = new Select(driver.findElement(By.name("newFundDirections.1.frontEndPercentage")));
+        FEOption2.selectByVisibleText("5%");
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 13]
```diff
@@ -0,0 +1,868 @@
...
+    @Then("^Extend maturity date validation$")
+    public void Extend_maturity_date_validation() {
+        objSmartSegPageElements = new SmartSegPageElements((RemoteWebDriver) driver);
+        WebElement Chkbx = driver.findElement(By.xpath("//*[@data-id='rdOneTimeDeposit']"));
+        WebElement parent = Chkbx.findElement(By.xpath("./.."));
+        parent.click();
+        objSmartSegPageElements.dateToday();
+        driver.findElement(By.xpath("//*[@data-id='txtTotalAmount']")).click();
+        objSmartSegPageElements.insertAmountFund();
+        Actions conChkbx = new Actions (driver);
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 14]
```diff
@@ -0,0 +1,868 @@
...
+        WebElement Chkbx2 = driver.findElement(By.xpath("//*[@data-id='chkBoxExtendMaturityDate']"));
+        conChkbx.moveToElement(Chkbx2, 0, 0).click().perform();
+        objSmartSegPageElements.ContinueBtnClick();
+        String message = objSmartSegPageElements.messageRequiredField();
+        Assert.assertEquals(message,"This field is required");
+        Actions radio = new Actions (driver);
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Actions class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 15]
```diff
@@ -0,0 +1,868 @@
...
+        WebElement datecalendar = driver.findElement(By.name("depositStartDate"));
+        Assert.assertEquals(true, datecalendar.isDisplayed());
+        Date dateToday = new Date();
+        SimpleDateFormat dateformat = new SimpleDateFormat("MMMM dd, YYYY");
+        String strDate= dateformat.format(dateToday);
+        driver.findElement(By.name("depositStartDate")).sendKeys(strDate);
+        driver.findElement(By.xpath("//*[@id='root']/div/div[3]/div/main/div[2]/div/div/div[2]/div/form/div[2]/div/section/div[2]/div[2]/div/table/thead/tr/th[4]")).click();
+        driver.findElement(By.name("depositAmount")).sendKeys("10000");
+        objSmartSegPageElements.addNewFundButtonClick();
+        Select fundClass = new Select(driver.findElement(By.name("newFundDirections.0.class")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 16]
```diff
@@ -0,0 +1,868 @@
...
+        fundClass.selectByVisibleText("C 100/100");
+        Select fundOption = new Select(driver.findElement(By.name("newFundDirections.0.feeOption")));
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 17]
```diff
@@ -0,0 +1,868 @@
...
+        fundOption.selectByVisibleText("Front-end");
+        Select fundCode = new Select(driver.findElement(By.name("newFundDirections.0.fundCode")));
+        fundCode.selectByVisibleText("CANADIAN EQUITY FUND");
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 18]
```diff
@@ -0,0 +1,868 @@
...
+        Select fundFF = new Select(driver.findElement(By.name("newFundDirections.0.frontEndPercentage")));
+        fundFF.selectByVisibleText("3%");
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: Select class move to the page factory

File: src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java [hunk 19]
```diff
@@ -0,0 +1,868 @@
...
+        Assert.assertEquals(driver.findElement(By.xpath("//*[@data-id='tblNewFundDirection']/tbody/tr[2]/td[2]/div")).getText(),"This field is required");
+    }
+
+    @Then("^Editing validations$")
+    public void editing_validations() throws InterruptedException {
+        // Edit email address
+        objSmartSegPageElements = new SmartSegPageElements((RemoteWebDriver) driver);
+        objSmartSegPageElements.BackBtnClick();
+        driver.findElements(By.className("icon-edit")).get(0).click();
+        driver.findElement(By.name("emailAddress")).sendKeys(Keys.BACK_SPACE);
```
Comment (ShiviPrabhakar) [inline src/main/java/com/glue/StepDef/Portal/SmartSeg/SmartSeg.java]: try using loops for this not the way to do it

File: resources/feature/Portal/Retail/AdvisorPortal/SmartSeg/smartseg.feature [hunk 20]
```diff
@@ -0,0 +1,489 @@
...
+    When  Confirmation Page validation
+    Then  eSignature page validation
+    Then  Log out
+    Examples:
+      | username | password  | policy    |
+      | A62523   | Empire100 | 401333703 |
+
+
+
+  Scenario Outline:PORTAL - PWS2017: Capped product message
```
Comment (ShiviPrabhakar) [inline resources/feature/Portal/Retail/AdvisorPortal/SmartSeg/smartseg.feature]: test failing please review

Comment (sunnysoomro12) [reply resources/feature/Portal/Retail/AdvisorPortal/SmartSeg/smartseg.feature]: I was working on this. I added data-id and removed previous locator. 
It is passing now.

**Comment by ShiviPrabhakar** [comment]
can be merged

**Pull request merged**